util/ecr.go

```diff
@@ -6,19 +6,24 @@ import (
 	"regexp"
 	"strings"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ecr"
 	"github.com/docker/docker/api/types"
+	v3 "github.com/rancher/rke/types"
 )
 
 const proxyEndpointScheme = "https://"
 
 var ecrPattern = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?`)
 
 // ECRCredentialPlugin is a wrapper to generate ECR token using the AWS Credentials
-func ECRCredentialPlugin(plugin map[string]string, pr string) (authConfig types.AuthConfig, err error) {
+func ECRCredentialPlugin(plugin v3.ECRCredentialPlugin, pr string) (authConfig types.AuthConfig, err error) {
```

Comment:
Why not use `*v3.ECRCredentialPlugin` as plugin parameter type instead of `v3.ECRCredentialPlugin` ??

Reply:
The original reason was there was no reason to change the plugin. I have anyways brought it in line with the rest of the pattern in the project.